MG-7 - Add User Documentation (#9)
absmach/magistrala-docs: docs/users-quick-start.md

---
title: Quick Start Guide
---

Magistrala leverages **SuperMQ** as the backbone for its **Services**, handling the creation, deletion, updating, and retrieval of user accounts, clients and channels.
Users in Magistrala must have **unique credentials**, including a `username`, `secret`, and `email` address upon creation.

This guide will take you through the core services for a quick set up of Magistrala and a walkthrough its services and components.

> After cloning and launching Magistrala, head to `localhost:3000` for the Magistrala UI platform.

Reviewer: remove this. We will be using cloud services so they won't need to clone and build

## **Sign Up**

> To create a new user with **standard permissions**, navigate to the **Register** page. The new user must provide:

Reviewer: ```suggestion
To get started, create an account by providing the following details in the sign up page: 
```

- **First and last name**

> - **A unique email address**

Reviewer: ```suggestion
- **An email address**
```

- **A username**

![Sign Up](../docs/img/users-guide/registeruser.png)

> Once registered, the user will receive a unique authentication token and be redirected to the **Domains Homepage**, where they can create and manage multiple domains.

Reviewer: no need of talking about the authentication token.

![Domain Homepage](../docs/img/users-guide/janedoe-domainshome.png)

> Tokens tend to expire after some time which will lead to the user being kicked out of the session. Do not be alarmed if this happens. Simply back log in through the Login page with the user's **username** or **email address** as well as password.

Reviewer: Remove this. It's not entirely correct

## **Log In**

> To access the platform, log in with your email or username and password.

Reviewer: ```suggestion
Incase you already have an account, you can log in with your email/username and password
```

![Login](../docs/img/users-guide/main-login.png)

## **Log into a Domain**

Upon logging in, users are redirected to the **Domain Selection Page**.

> A **Domain** is a workspace that contains **Clients**, **Channels**, **Groups**, **Dashboards** as well as **Roles** and **Rules Engine** Services. A user can create as many domains as they please. A **Super Admin** has access to all domains present in the system. But a **User** must be a member or creator of a domain to log into it.

Reviewer: ```suggestion
A **Domain** is a workspace that allows you to manage **Clients**, **Channels**, **Groups**, **Dashboards**, **Members**,  **Rules**  and **Bootstrap** services. A user can create as many domains as they please. 
```

Remove all super admin related talk in any other sections.

Click on the `+ Create` button on the top right to create a new domain. Since multiple domains can have the same name, you must add an **alias** which will be a unique descriptor for the domain.

![Domain Create](../docs/img/users-guide/jdoe-create-domain.png)

> Once you create a domain, you are given **Administrator** permission over the domain. You are able to perform all actions available over the domain and all the entities provisioned inside the domain. You can also assign or invite members to the domain with various levels of permissions.

Reviewer: ```suggestion
Once you create a domain, you are given **admin** role over the domain by default. You are able to perform all actions available over the domain and all the entities provisioned inside the domain. You can also assign or invite members to the domain with various levels of permissions.
```

You also need to tell the user what they should do after creating the domain, they click on the card to log into the domain
